### ISSUE TITLE
no-fallthrough broken on nested switch statements

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#430]
The `no-fallthrough` rule fails on this pattern:

``` js
switch(foo) {
    case 0: 
        switch(bar) { 
            case 1: break; 
            case 2: break 
        } 
       /* falls through */ 
   case 3: return
}
```

It flags `case 0` as not having a proper ending.
